mne/coreg.py

```diff
@@ -1223,3 +1227,586 @@ def _scale_xfm(subject_to, xfm_fname, mri_name, subject_from, scale,
+            if self._n_scale_param == 0:
+                self._parameters[:6] = est
+            elif self._n_scale_param == 1:
+                self._parameters[:] = list(est) + [est[-1]] * 2
+            else:
+                self._parameters[:] = est
+            angle, move, scale = self._changes
+            self._parameters_items_changed()
+            if angle <= self._icp_angle and move <= self._icp_distance and \
+                    all(scale <= self._icp_scale):
+                break
+
+    def omit_hsp_points(self, distance):
+        """Exclude head shape points that are far away from the MRI head.
+
+        Parameters
+        ----------
+        distance : float
+            Exclude all points that are further away from the MRI head than
+            this distance. A value of distance <= 0 excludes nothing.
+        """
+        distance = float(distance)
+        if distance <= 0:
+            return
+
+        # find the new filter
+        mask = self._orig_hsp_point_distance <= distance
+        n_excluded = np.sum(~mask)
+        logger.info("Coregistration: Excluding %i head shape points with "
+                    "distance >= %.3f m.", n_excluded, distance)
+        # set the filter
+        with warnings.catch_warnings(record=True):  # comp to None in Traits
+            self._hsp.points_filter = mask
+
+    def point_distance(self):
+        """Compute Euclidean distance between the head-mri points."""
+        mri_points = list()
+        hsp_points = list()
+        if self._hsp_weight > 0 and self._has_hsp_data:
+            mri_points.append(self._transformed_high_res_mri_points[
+                self._nearest_transformed_high_res_mri_idx_hsp])
+            hsp_points.append(self._transformed_hsp_points)
+            assert len(mri_points[-1]) == len(hsp_points[-1])
+        if self._eeg_weight > 0 and self._has_eeg_data:
+            mri_points.append(self._transformed_high_res_mri_points[
+                self._nearest_transformed_high_res_mri_idx_eeg])
+            hsp_points.append(self._transformed_hsp_eeg_points)
+            assert len(mri_points[-1]) == len(hsp_points[-1])
+        if self._hpi_weight > 0 and self._has_hpi_data:
+            mri_points.append(self._transformed_high_res_mri_points[
+                self._nearest_transformed_high_res_mri_idx_hpi])
+            hsp_points.append(self._transformed_hsp_hpi)
+            assert len(mri_points[-1]) == len(hsp_points[-1])
+        if all(len(h) == 0 for h in hsp_points):
+            return None
+        mri_points = np.concatenate(mri_points)
+        hsp_points = np.concatenate(hsp_points)
+        return np.linalg.norm(mri_points - hsp_points, axis=-1)
+
+    def save_trans(self, fname):
+        """Save the head-mri transform as a fif file.
+
+        Parameters
+        ----------
+        fname : str
+            Target file path.
+        """
+        write_trans(fname, Transform('head', 'mri', self._head_mri_t))
```

Comment:
We don't really need this in the class. Instead we can have a `.trans` property that returns the `Transform(...)`, and in the actual GUI we can just call `write_trans(fname, coreg.trans)`

Reply:
+1